src/basic/unit-file.c

```diff
@@ -412,15 +413,31 @@ int unit_file_build_name_map(
                 if (null_or_empty_path(dst) != 0)
                         continue;
 
-                /* Do not treat instance symlinks that point to the template as aliases */
-                if (unit_name_is_valid(basename(dst), UNIT_NAME_TEMPLATE) &&
-                    unit_name_is_valid(src, UNIT_NAME_INSTANCE))
-                        continue;
+                dst = basename(dst);
+
+                /* If we have an symlink from an instance name to a template name, it is an alias just for
+                 * this specific instance, foo@id.service ↔ template@id.service. */
+                if (unit_name_is_valid(dst, UNIT_NAME_TEMPLATE)) {
+                        UnitNameFlags t = unit_name_to_instance(src, &inst);
```

Comment:
hmm

unit_name_to_instance() returns an `int`, not a UnitNameFlags?

Reply:
`UnitNameFlags` is the same as `int`, so there is no functional change. I changed this so gdb shows the enum name when printing. In the past we left `unit_name_is_valid()` to return `int`, but I think it makes more sense to change it to return `UnitNameFlags`. (Esp. that `_UNIT_NAME_INVALID == -EINVAL` now). I'll submit that as a separate PR.

Reply:
I now prepped a patch to do that. I'll submit it once this is merged, to avoid conflicts.